### ISSUE TITLE
Question about schemes within properties

### ISSUE THREAD

**Issue by ogiel** [RealEstateCore/rec#236]
We are currently working with the most recent version of REC from this repo but are having a few issues / questions about DTDL behavior (version: [https://github.com/RealEstateCore/rec/blob/main/RealEstateCore.DTDLv2.jsonld](https://github.com/RealEstateCore/rec/blob/main/RealEstateCore.DTDLv2.jsonld))

For example the model with id: "dtmi:org:brickschema:schema:Brick:Alarm;1" contains a component with name lastKnownValue which has a schema of "dtmi:org:w3id:rec:ExceptionEvent;1". When we open the ADT Explorer we can see that the Alarm contains this lastKnownValue component with two properties, value and timestamp.

However, when we dig into the ExceptionEvent component it eventually extends "dtmi:org:w3id:rec:Event;1" (full drill down below). The event model contains quite some properties like a customProperties map, but also event ending timestamp and event start timestamps. The very last property it contains is the timestamp, which is the only one we do actually see.

Could someone elaborate on this so it is clear to us why this is expected, and if not what we might have done wrong? 

dtmi:org:brickschema:schema:Brick:Alarm;1
lastKnowValue has schema -> 
dtmi:org:w3id:rec:ExceptionEvent;1
has property "value" and extends ->
dtmi:org:w3id:rec:PointEvent;1
extends ->
dtmi:org:w3id:rec:Event;1
has property "customProperties",  "customTags", "end", "identifiers", "name", "start" and "timestamp"

**Comment by cbupp** [comment]
@ogiel - not sure I follow. The below screenshot is from ADT Explorer:

![image](https://user-images.githubusercontent.com/3836596/232135695-067a0f57-a18d-422c-8c13-72e4a8c561aa.png)

I can see all of the inherited properties.

**Comment by WesVerhagen** [comment]
Are you on dtdl v3?

**Comment by WesVerhagen** [comment]
@cbupp 
We only have the following properties.
![Screenshot 2023-04-17 092645](https://user-images.githubusercontent.com/95536067/232414680-94acd9c4-1450-4047-88be-214176f5055b.png)

But like we said in the following issue:
https://github.com/RealEstateCore/rec/issues/237 
We used ADT import jobs because we got a error using the GUI in the explorer.

I will try to use the tool with the batch option and will come back to this issue.

**Comment by WesVerhagen** [comment]
@cbupp 
Okey i have tried to use the tool with the batch (-b 10) option and the upload was a success.
unfortunately, we get the following error when we try to open the lastKnowValue on a Alarm twin.

![image](https://user-images.githubusercontent.com/95536067/232435855-77956c1f-3584-49a5-a695-2bb56c839e2c.png)

Do you know what is going on and how to fix it?

**Comment by cbupp** [comment]
> Are you on dtdl v3?

@ogiel - I'm not sure I follow. The REC schemas are published to DTDLv2. Have you updated these to be DTDLv3?

**Comment by ogiel** [comment]
Hey I'll try to give an exact description of what we did and what the results are:

First we download the [RealEstateCore.DTDLv2.jsonld](https://github.com/RealEstateCore/rec/blob/main/RealEstateCore.DTDLv2.jsonld) file.

Then we do one of the following:
1. Change the format to .json so ADT accepts it, then we use the UI to upload models, this results in an error but some models do get uploaded.
2. Push the .jsonld file through import jobs API. This results in a success but then we do not see the full list of extended properties (on which the original question was based).
3. As you explained how you do it; we downloaded the ADT tools script and pushed the REC files from the directories with a batch size of 10, but this results in the error from @WesVerhagen his last message; Cannot read properties of undefined.

Imo two strange things are happening here, the first is that we have not found a robust solution and the second that all these options that should have the same result all have different outcomes. Maybe someone from the ADT team could help out?

**Comment by ogiel** [comment]
@cbupp Ok after retrying and verifying everything we have found the issues and they are on our side. We have used version 4.0.0.17 instead of 4.0.0.18 for the RealEstateCore.DTDLv2.jsonld file. After cleaning up a full instance, retransforming the jsonld into ndjson for import jobs and then recreating the twins we do get it to work.

This means that 2. and 3. from the previous message can be discarded and sorry for this inconvenience. The UI with batching issues remains so we will just use import jobs.
